### PR TITLE
New dnsdist documentation

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description

This revamps the dnsdist documentation to use restructured text and sphinx as build system.

Comments on theme, content, style, structure and anything else are welcome.

The current version of these built documentation are available [here](http://dnsdist.org/new/), as is [the pdf](http://dnsdist.org/new/dnsdist.pdf).

This site is meant to replace the main dnsdist website in due time.

ToDo:

* Make the README.md a true, short readme
* Create the changelog with all changes between 1.1.0 and the upcoming 1.2.0
* Split up the reference guide into more separate files
* Make the builder upload the documentation

Future work:

* More guides

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by giganteous** [comment]
The documentation for the function 'newRemoteLogger' is missing.

**Comment by giganteous** [comment]
Browsing directly to http://dnsdist.org/new/statistics.html#cache-hits (or clicking on 'cache-hits') moves the left side (toc) off of my screen.

**Comment by pieterlexis** [comment]
> Browsing directly to http://dnsdist.org/new/statistics.html#cache-hits (or clicking on 'cache-hits') moves the left side (toc) off of my screen.

Can you share a screenshot of this?

File: pdns/dnsdistdist/docs/advanced/ecs.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Using ENDS Client Subnet
```
Comment (rgacogne) [inline pdns/dnsdistdist/docs/advanced/ecs.rst]: EDNS

**Comment by giganteous** [comment]
The advanced section could use a TeeAction section copied in from the original docs; getting stats from it was lost in reStructuring?

**Pull request merged**